examples/advanced/xgboost/fedxgb_secure/README.md

```diff
@@ -68,47 +68,30 @@ bash run_training_standalone.sh
+> The actual encyption plugin will be used in the next step.
 
-```
-# config the job template directory
-nvflare config -jt ../../../job_templates/
-
-# create horizontal job
-nvflare job create -force -w xgboost -j ./jobs/xgb_hori \
-    -f config_fed_server.conf secure_training=false data_split_mode=0 \
-    -f config_fed_client.conf folder="/tmp/nvflare/xgb_dataset/horizontal_xgb_data"
-
-# create horizontal secure job
-nvflare job create -force -w xgboost -j ./jobs/xgb_hori_secure \
-    -f config_fed_server.conf secure_training=true data_split_mode=0 \
-    -f config_fed_client.conf folder="/tmp/nvflare/xgb_dataset/horizontal_xgb_data"
-
-# create vertical job
-nvflare job create -force -w xgboost -j ./jobs/xgb_vert \
-    -f config_fed_server.conf secure_training=false data_split_mode=1 \
-    -f config_fed_client.conf folder="/tmp/nvflare/xgb_dataset/vertical_xgb_data"
-
-# create vertical secure job
-nvflare job create -force -w xgboost -j ./jobs/xgb_vert_secure \
-    -f config_fed_server.conf secure_training=true data_split_mode=1 \
-    -f config_fed_client.conf folder="/tmp/nvflare/xgb_dataset/vertical_xgb_data"
-
-```
+## Federated Experiments with NVFlare
+We then run the federated XGBoost training using NVFlare Simulator via [JobAPI](https://nvflare.readthedocs.io/en/main/programming_guide/fed_job_api.html), without and with homomorphic encryption.
+The running time of each job depends mainly on the encryption workload. 
 
-Or you can just run the script:
 ```
-bash prepare_flare_job.sh
+python xgb_fl_job.py --data_root /tmp/nvflare/dataset/xgb_dataset/horizontal_xgb_data --data_split_mode horizontal
+python xgb_fl_job.py --data_root /tmp/nvflare/dataset/xgb_dataset/horizontal_xgb_data --data_split_mode horizontal --secure True
+python xgb_fl_job.py --data_root /tmp/nvflare/dataset/xgb_dataset/vertical_xgb_data --data_split_mode vertical
+python xgb_fl_job.py --data_root /tmp/nvflare/dataset/xgb_dataset/vertical_xgb_data --data_split_mode vertical --secure True
 ```
 
-## Run Federated Experiments with NVFlare
-Next, we run the federated XGBoost training without and with homomorphic encryption using NVFlare. 
-We run the NVFlare jobs using simulator with: 
+Due to the design that secure horizontal scheme will perform secure aggregation at server-side, we need to provision additional 
+tenseal context before starting the job to get server prepared (in contrast, secure vertical scheme does not have such needs
+because server only performs message routing without doing the actual secure message aggregation). The following steps are needed
+to continue the secure horizontal job experiment:
 ```
-bash run_training_flare.sh
+jobdir=/tmp/nvflare/workspace/fedxgb_secure/train_fl/jobs/horizontal_secure
+workdir=/tmp/nvflare/workspace/fedxgb_secure/train_fl/works/horizontal_secure
+nvflare provision -p project.yml -w ${workdir}
+nvflare simulator ${jobdir} -w ${workdir}/example_project/prod_00/site-1 -n 3 -t 3
 ```
```

Comment:
Thanks for adding the explanation, could we add one or two lines about the HE builder and project.yml for example:

"""
In the secure horizontal scheme, secure aggregation is performed on the server-side. To support this, additional tenseal context must be provisioned before starting the job to prepare the server. In contrast, the secure vertical scheme doesn't require this step because the server's role is limited to message routing, without performing the actual secure message aggregation.

To enable this, we have implemented the nvflare.lighter.impl.he.HEBuilder, which is included in the prepared project.yml. You can provision the server by running the following commands:
"""

Reply:
updated